frontend/src/routes/Home.svelte

```diff
@@ -1,22 +1,178 @@
 <script lang="ts">
-</script>
+  import yup from 'yup';
+  import svelteFormsLib from 'svelte-forms-lib';
+  import { _ } from 'svelte-i18n';
+  import { navigate } from 'svelte-routing';
+  import Spinner from 'svelte-tailwind-widgets/Spinner.svelte';
+  import { createConfiguration, DefaultApi, ServerConfiguration } from 'jimmi-api-client/mod.ts';
+  import config from '../config.ts';
+
+  interface IJimmiSettings {
+    domain: string;
+    room: string;
+    password: string;
+  }
+
+  const domainRegex = new RegExp(
+    /^$|^((?:(?:(?:\w[\.\-\+]?)*)\w)+)((?:(?:(?:\w[\.\-\+]?){0,62})\w)+)\.(\w{2,6})$/
+  );
+
+  const jitsiRoomRegex = new RegExp(/^[^?&:"'%#]+$/); // only given characters are excluded
+
+  const { form, errors, handleChange, handleSubmit } = svelteFormsLib.createForm({
+    initialValues: {
+      domain: '',
+      room: '',
+      password: '',
+    },
+    validationSchema: yup.object().shape({
+      domain: yup.string().trim().matches(domainRegex),
+      room: yup.string().trim().min(1).matches(jitsiRoomRegex).required(),
+      password: yup.string().optional(),
+    }),
+    onSubmit: (settings: IJimmiSettings) => {
+      if (!settings.domain) {
+        settings.domain = config.defaultJitsiUrl;
+      }
+      isLoading = true;
+      createJimmiInstance(settings);
+    },
+  });
+
+  /**
+   * Creates a new Jimmi instance and redirects the user to the corresponding dashboard
+   *
+   * @param settings - The Jimmi instance settings
+   */
+  async function createJimmiInstance(settings: IJimmiSettings) {
+    const apiConfig = createConfiguration({
+      baseServer: new ServerConfiguration(config.apiBaseUrl),
+    });
+    const jimmiApi = new DefaultApi(apiConfig);
+    const instance = await jimmiApi.instancesPost();
+    // ToDo: Join room etc, once API is ready for that
+    navigate(`/instance/${instance.id}`);
+  }
 
-<h1>Hello World!</h1>
-<p>
-  Visit the <a href="https://svelte.dev/tutorial">Svelte tutorial</a> to learn how
-  to build Svelte apps.
-</p>
-
-<link
-  href="https://fonts.googleapis.com/css?family=Overpass:100,400"
-  rel="stylesheet"
-/>
-
-<style>
-  h1 {
-    color: #ff3e00;
-    text-transform: uppercase;
-    font-size: 4em;
-    font-weight: 100;
+  /**
+   * EventHandler for onBeforeInput event. If a complete jitsi url is pasted into the input field,
+   * the domain and conference are extracted automatically.
+   *
+   * @param event - The onBeforeInput event
+   */
+  function beforeInput(event: Event) {
+    if (event instanceof InputEvent && event.data && event.inputType === 'insertFromPaste') {
+      const { data } = event;
+      const urlRegex = new RegExp('^http(s?)://(.*)/(.*)$');
```

Comment:
Should we support automatically inserting of domains with a trainling slash too?

Reply:
Not quite sure about the benefit of that. This functionality is used to split a copied Jitsi conference url, which never ends with a slash by default